diff --git a/CMakeLists.txt b/CMakeLists.txt
index 846b444..b1fb31a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -79,7 +79,7 @@ if (WITH_GMOCK AND TARGET GTest::gmock)
 endif (WITH_GMOCK AND TARGET GTest::gmock)
 
 if (WITH_GFLAGS)
-  find_package (gflags 2.2.2)
+  # find_package (gflags 2.2.2)
 
   if (gflags_FOUND)
     set (HAVE_LIB_GFLAGS 1)
@@ -987,12 +987,12 @@ if (BUILD_TESTING)
   set_tests_properties (cleanup_with_relative_prefix PROPERTIES FIXTURES_REQUIRED logcleanuptest)
 endif (BUILD_TESTING)
 
-install (TARGETS glog
-  EXPORT glog-targets
-  RUNTIME DESTINATION ${_glog_CMake_BINDIR}
-  PUBLIC_HEADER DESTINATION ${_glog_CMake_INCLUDE_DIR}/glog
-  LIBRARY DESTINATION ${_glog_CMake_LIBDIR}
-  ARCHIVE DESTINATION ${_glog_CMake_LIBDIR})
+# install (TARGETS glog
+#   EXPORT glog-targets
+#   RUNTIME DESTINATION ${_glog_CMake_BINDIR}
+#   PUBLIC_HEADER DESTINATION ${_glog_CMake_INCLUDE_DIR}/glog
+#   LIBRARY DESTINATION ${_glog_CMake_LIBDIR}
+#   ARCHIVE DESTINATION ${_glog_CMake_LIBDIR})
 
 if (WITH_PKGCONFIG)
   install (
@@ -1025,8 +1025,8 @@ write_basic_package_version_file (
   ${CMAKE_CURRENT_BINARY_DIR}/glog-config-version.cmake
   COMPATIBILITY SameMajorVersion)
 
-export (TARGETS glog NAMESPACE glog:: FILE glog-targets.cmake)
-export (PACKAGE glog)
+# export (TARGETS glog NAMESPACE glog:: FILE glog-targets.cmake)
+# export (PACKAGE glog)
 
 get_filename_component (_PREFIX "${CMAKE_INSTALL_PREFIX}" ABSOLUTE)
 
@@ -1076,5 +1076,5 @@ install (DIRECTORY ${_glog_BINARY_CMake_DATADIR}
   FILES_MATCHING PATTERN "*.cmake"
 )
 
-install (EXPORT glog-targets NAMESPACE glog:: DESTINATION
-  ${_glog_CMake_INSTALLDIR})
+# install (EXPORT glog-targets NAMESPACE glog:: DESTINATION
+#   ${_glog_CMake_INSTALLDIR})
